src/components/Utility/filterFeatChart.js

```diff
@@ -0,0 +1,107 @@
+export const filterFeatChart = (data, selectedFeat) => {
+  const { selectedfilterFeat, selectedfilterSubfeat, selectedCounty, featLabel } = selectedFeat
+  let pieData = []
+  let reqCountyData = (data && selectedCounty) && data[selectedCounty.id]
+
+  if (reqCountyData) {
+
+    switch(selectedfilterFeat) {
+
+      case 'poverty_data':
```

Comment:
I think we can similarly make this more stable by using objects instead of arrays. When you're referencing something by index, very small changes can break everything. Let me know if you have any questions about implementing this!